honeyscanner/detect_honeypot/detect_honeypot.py

```diff
@@ -0,0 +1,283 @@
+    message = b'\x00\x00\x00\xa4\xff\x53\x4d\x42\x72\x00\x00\x00\x00\x08\x01\x40\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x40\x06\x00\x00\x01\x00\x00\x81\x00\x02PC NETWORK PROGRAM 1.0\x00\x02MICROSOFT NETWORKS 1.03\x00\x02MICROSOFT NETWORKS 3.0\x00\x02LANMAN1.0\x00\x02LM1.2X002\x00\x02Samba\x00\x02NT LANMAN 1.0\x00\x02NT LM 0.12\x00'
+    resp = connect_to_socket(ip, 445, message)
+    default_banner = b'SMBr'
+    if default_banner in resp:
+        print('[\033[92m+\033[00m] Found dionaea smb server')
+        return True
+    else:
+        print('[\033[91m-\033[00m] Didn\'t find dionaea smb server')
+        return False
+
+
+def check_cowrie(ip):
+    """
+    Checks if a given IP address is running a Cowrie honeypot by connecting via SSH and performing tests.
+
+    Args:
+        ip (str): The IP address of the host to check.
+
+    Returns:
+        None: This function performs checks and does not return a value.
+    """
+    ssh_conn = SSHConnectCowrie(ip=ip, port=2222, username="root", password="1234")
+    ssh_conn.connect()
+    ssh_conn.check_os_version()
+    ssh_conn.check_meminfo()
+    ssh_conn.check_mounts()
+    ssh_conn.check_cpu()
+    ssh_conn.check_group()
+    ssh_conn.check_shadow()
+    ssh_conn.check_passwd()
+    ssh_conn.check_hostname()
+    ssh_conn.close()
+
+
+def check_dionaea(ip):
+    """
+    Checks if a given IP address is running a Dionaea honeypot by checking specific ports.
+
+    Args:
+        ip (str): The IP address of the host to check.
+
+    Returns:
+        bool: True if the responses from the ports indicate a Dionaea honeypot, False otherwise.
+    """
+    res_21 = port_21(ip)
+    res_443 = port_443(ip)
+    res_445 = port_445(ip)
+    return res_21 and res_443 and res_445
+
+
+def detect_honeypot(ip):
+    """
+    Detects if a given IP address is running a known honeypot based on open ports and responses.
+
+    Args:
+        ip (str): The IP address of the host to check.
+
+    Returns:
+        None: This function prints the detection results.
+    """
+    open_ports = check_open_ports(ip)
+
+    port_cowrie_kippo = 2222
+    dionaea_ports = [21, 23, 42, 53, 80, 135, 443, 445, 1433, 1723, 1883, 3306, 5060, 9100, 11211, 27017]
+    conpot_ports = [2121, 5020, 10201, 44818]
+
+    if port_cowrie_kippo in open_ports:
+        resp = port_2222(ip)
+        if resp == 'cowrie':
+            check_cowrie(ip)
+            print('This host is probably Cowrie honeypot')
+            version = get_latest_version('cowrie')
+            print(f'Cowrie version {version}')
+        elif resp == 'kippo':
+            print('This host is probably Kippo honeypot')
+            version = get_latest_version('kippo')
+            print(f'Kippo version {version}')
+    elif set(dionaea_ports) <= set(open_ports):
+        print('The same open ports as in Dionaea honeypot')
+        if check_dionaea(ip):
+            print('This host is probably Dionaea honeypot')
```

Comment:
Add one more printing describing the reason, that you tried connecting to 3 ports and was successful.